Fix broken links in the docs

If you run `make linkcheck` in the docs, there are several links that are broken and should be fixed. There's also a bunch of permanent redirects that can be updated.  

There are also several links to `docs.sympy.org`. These should be replaced with relative links using Sphinx (it would be nice if there were an automated check for this).

When I try to run `make linkchekc` at ~/doc directory it returns the following error:
`~/doc/make linkcheck
mkdir -p _build/linkcheck _build/doctrees
sphinx-build -b linkcheck -d _build/doctrees  src _build/linkcheck
Running Sphinx v8.1.3

Configuration error:
There is a programmable error in your configuration file:

Traceback (most recent call last):
  File "/home/matheus/Codes/Projetos Pessoais/Contribuições Projetos Open Source/sympy/venv/lib/python3.12/site-packages/sphinx/config.py", line 529, in eval_config_file
    exec(code, namespace)  # NoQA: S102
    ^^^^^^^^^^^^^^^^^^^^^
  File "/home/matheus/Codes/Projetos Pessoais/Contribuições Projetos Open Source/sympy/doc/src/conf.py", line 19, in <module>
    from intersphinx_registry import get_intersphinx_mapping
ModuleNotFoundError: No module named 'intersphinx_registry'

make: *** [Makefile:125: linkcheck] Error 2
`

Am I doing something wrong? At first I though it could be that sphinx wasn't installed at sympy folder and did it through venv, however the error still continues.

When I try to run `make linkchekc` at ./doc directory it returns the following error:

$make linkcheck
mkdir -p _build/linkcheck _build/doctrees
sphinx-build -b linkcheck -d _build/doctrees  src _build/linkcheck
Running Sphinx v8.1.3

Configuration error:
There is a programmable error in your configuration file:

Traceback (most recent call last):
  File "./sympy/venv/lib/python3.12/site-packages/sphinx/config.py", line 529, in eval_config_file
    exec(code, namespace)  # NoQA: S102
    ^^^^^^^^^^^^^^^^^^^^^
  File "./sympy/doc/src/conf.py", line 19, in <module>
    from intersphinx_registry import get_intersphinx_mapping
ModuleNotFoundError: No module named 'intersphinx_registry'

make: *** [Makefile:125: linkcheck] Error 2


Am I doing something wrong? At first I though it could be that sphinx wasn't installed at sympy folder and did it through venv, however the error still continues.

--- a/.mailmap
+++ b/.mailmap
@@ -1396,6 +1396,7 @@ Safiya03 <safiyanesar@gmail.com>
 Sagar Bharadwaj <sagarbharadwaj50@gmail.com>
 Sagar231 <sagarfeb298@gmail.com>
 Sahil Shekhawat <sahilshekhawat01@gmail.com>
+Sai Kiran P <parnandi.saikiran@gmail.com> saikiranp-24 <parnandi.saikiran@gmail.com>
 Sai Nikhil <tsnlegend@gmail.com>
 Sai Udayagiri <saibabu.udayagiri@gmail.com> <144413200+saiudayagiri@users.noreply.github.com>
 Sai Udayagiri <saibabu.udayagiri@gmail.com> Sai Udayagiri <saibabu.udayagiri@email.com>

--- a/doc/src/contributing/dependencies.md
+++ b/doc/src/contributing/dependencies.md
@@ -153,7 +153,7 @@ solvers if they are installed. Note that `satisfiable()` is also used by
 
 - **pysat**: [Pysat](https://pysathq.github.io/) is a library which wraps many
   SAT solvers. It can also be used as a backend to `satisfiable()`. Presently,
-  only [Minisat](http://minisat.se/MiniSat.html) is implemented, using
+  only [Minisat](https://github.com/niklasso/minisat) is implemented, using
   `satisfiable(algorithm=minisat22')`.
 
 ### Plotting

